x-pack/plugins/security_solution/public/timelines/components/timeline/session_tab_content/use_session_view.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useCallback } from 'react';
+import { EuiButtonEmpty, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import type { EntityType } from '../../../../../../timelines/common';
+import { timelineSelectors } from '../../../store/timeline';
+import { useKibana } from '../../../../common/lib/kibana';
+import { TimelineId, TimelineTabs } from '../../../../../common/types/timeline';
+import { timelineDefaults } from '../../../../timelines/store/timeline/defaults';
+import { useDeepEqualSelector } from '../../../../common/hooks/use_selector';
+import { useDetailPanel } from '../../side_panel/hooks/use_detail_panel';
+import { SourcererScopeName } from '../../../../common/store/sourcerer/model';
+import { isFullScreen } from '../body/column_headers';
+import {
+  SCROLLING_DISABLED_CLASS_NAME,
+  FULL_SCREEN_TOGGLED_CLASS_NAME,
+} from '../../../../../common/constants';
+import { FULL_SCREEN } from '../../timeline/body/column_headers/translations';
+import { EXIT_FULL_SCREEN } from '../../../../common/components/exit_full_screen/translations';
+import {
+  useTimelineFullScreen,
+  useGlobalFullScreen,
+} from '../../../../common/containers/use_full_screen';
+import {
+  updateTimelineGraphEventId,
+  updateTimelineSessionViewSessionId,
+  setActiveTabTimeline,
+} from '../../../../timelines/store/timeline/actions';
+import { detectionsTimelineIds } from '../../../containers/helpers';
+import * as i18n from './translations';
+
+const FullScreenButtonIcon = styled(EuiButtonIcon)`
+  margin: 4px 0 4px 0;
+`;
+interface NavigationProps {
+  fullScreen: boolean;
+  globalFullScreen: boolean;
+  onCloseOverlay: () => void;
+  timelineId: TimelineId;
+  timelineFullScreen: boolean;
+  toggleFullScreen: () => void;
+  graphEventId?: string;
+  activeTab: TimelineTabs;
+}
+
+const NavigationComponent: React.FC<NavigationProps> = ({
```

Comment:
We should separate each of these into their own files when there's time and create test for each of them